### PR TITLE
feat: Breadcrumb class + StarterBase auto-context (next release)

### PR DESCRIPTION
**Pull request by parisek**
## Summary

- New `Parisek\TimberKit\Breadcrumb` class — strategy dispatcher returning typed item arrays for any WP query state (404, search, date, author, archives, taxonomy, singular, pagination)
- `StarterBase` configuration surface: 5 new `$breadcrumb_*` properties + auto-populate of `$context['breadcrumb']` in `timber_context()`
- Legacy compatibility guard: `class_exists('\Breadcrumb', false)` skips auto-populate when consuming project still ships old global class
- Filter API: `timber_kit_breadcrumb_items|labels|skip|menu_trail` (matches existing `timber_kit_*` naming convention)
- ~40 brain/monkey tests covering all strategy branches + helpers + filters + edge cases

## Why a spec-first workflow

The full design lives in `docs/adr/2026-05-24-breadcrumb-design.md` (copied from the consuming `wordpress-base` project where this work originated). 14 sections + §4.1 legacy guard sub-section. Implementation tasks were tracked against the spec via TDD.

## Migration impact

**Backward compatible** for downstream consumers — legacy `\Breadcrumb` class detection auto-opts-out of new auto-populate. See `CHANGELOG.md` `[Unreleased]` Migration note for the two-path guide.

The items shape `[{type, url, title}]` is shape-compatible with previous `[{url, title}]` — `type` is an additive key, no Twig component changes required downstream.

## Test plan

- [ ] `composer test` — all 656 tests pass (existing + new ~42 breadcrumb tests)
- [ ] `composer phpstan` — 0 errors
- [ ] Review the ADR (`docs/adr/2026-05-24-breadcrumb-design.md`) against the implementation
- [ ] Spot-check that existing `Helpers::*` / `Resizer` / `BlockRenderer` tests still pass (no regression)
- [ ] Confirm filter naming `timber_kit_breadcrumb_*` matches your convention preference
- [ ] Confirm the `[Unreleased]` CHANGELOG entry slot (bundled with speculation-rules work)
- [ ] Mark PR "Ready for review" when satisfied, then merge + tag (likely `v1.7.0`)

## Follow-up

After this PR is merged + tagged, a separate PR against `portadesign/wordpress-base` will:
- `composer require parisek/timber-kit:^1.7` (or whatever the merged tag becomes)
- Delete project-side `classes/Breadcrumb.php` + tests entirely
- Refactor `Base.php` to use `$breadcrumb_*` property overrides

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/StarterBase.php [hunk 1]
```diff
@@ -592,6 +613,27 @@ public function timber_context( $context ) {
...
+		// a global \Breadcrumb class (legacy convention from wordpress-base
+		// versions before 1.6.0). The class_exists check (with autoload=false)
+		// is a plain class-map lookup, not a filesystem probe; if the project
+		// autoloads classes/Breadcrumb.php via Composer, Base.php's constructor
+		// has already loaded it by the time timber_context() fires.
+		//
+		// Skipping when legacy class exists avoids wasted compute — the
+		// project's Base::timber_context() overwrites $context['breadcrumb']
+		// anyway via its own `new \Breadcrumb()` call.
+		if ( ! class_exists( '\Breadcrumb', false ) ) {
```
[the remark this answers is not in the document]
Comment (parisek) [reply src/StarterBase.php]: Fixed in c204532 — switched to `class_exists("\Breadcrumb")` (autoload=true default). Now reliably detects the legacy class via Composer classmap regardless of when it was previously instantiated. Comment block updated to remove the stale `autoload=false` rationale.

File: src/Breadcrumb.php [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+				'url'   => (string) $menu_item->url,
+			];
+		}
+
+		/** @var array<int, array<string, string>> $breadcrumb_items */
+		$breadcrumb_items = apply_filters(
+			'timber_kit_breadcrumb_menu_trail',
+			$breadcrumb_items,
+			$menu_name
+		);
```
[the remark this answers is not in the document]
Comment (parisek) [reply src/Breadcrumb.php:373]: Fixed in c204532 — renamed the local var to `$resolved_menu` so `$this->menu_name` stays untouched. The filter now always receives the original string `$this->menu_name` (config location name) regardless of whether `get_nav_menu_locations()` resolved it to an int menu ID for the internal lookup.

File: docs/adr/2026-05-24-breadcrumb-design.md [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+The project's `classes/Breadcrumb.php` is **deleted entirely**. `wordpress-base` is a create-project template (canonical source new projects clone from), not a long-running site — there are no historical `new \Breadcrumb()` call sites that need a graceful migration path. The class moves upstream cleanly. (See §9 for the rationale and the one-line `Base::timber_context()` migration.)
+
+---
+
+## 2. Scope
+
+| Repo | Changes | Version |
+|---|---|---|
+| **`parisek/timber-kit`** | `src/Breadcrumb.php` (new), `src/StarterBase.php` (+5 properties + auto-context behind legacy-class guard), `tests/Unit/BreadcrumbTest.php` (~35 brain/monkey tests), `CHANGELOG.md` | 1.6.0 (minor, additive) |
+| **`wordpress-base`** | `classes/Base.php` (use `$breadcrumb_*` properties), **delete** `classes/Breadcrumb.php`, **delete** `tests/unit/BreadcrumbTest.php`, `.claude/rules/wordpress/timber-kit.md` (new property rows) | n/a |
```
[the remark this answers is not in the document]
Comment (parisek) [reply docs/adr/2026-05-24-breadcrumb-design.md:28]: Fixed in 7e30597 — 8 instances of TimberKit version refs across 7 distinct locations replaced with Unreleased / next release / new tag depending on context. Refs like Timber 2.0+, PHP 8.3+, PHPUnit 11/12 and Composer constraint examples left unchanged (do not refer to the breadcrumb TimberKit version). Also addressed the related ADR spec/impl mismatch on line 227 (flat-CPT row) in the same commit.
